### ISSUE TITLE
Should the distribution of boxes be reproducible?

### ISSUE THREAD

**Issue by nncarlson** [AMReX-Codes/amrex#500]
Should the distribution of boxes be reproducible from one run to the next?  -- same mesh description, same number of mpi ranks.

I've encountered a case where it is occasionally not the same. I'm actually doing a single run consisting of a sequence of simulations, each enclosed in an `amrex_init`/`amrex_finalize` pair (Fortran interface, btw). Each sets up exactly the same grid (single level, single starting box). It looks like I always get the same set of boxes, but occasionally some boxes will be assigned to different mpi ranks.

**Comment by nncarlson** [comment]
Some more info. This is a small problem with 18 boxes (all identically sized). The distribution always seems to be reproducible when the number of ranks  evenly divides 18; i.e., 2, 3, 6, and 9. But for any other number of ranks, the distribution is non-reproducible (seems never to be in fact).

**Comment by WeiqunZhang** [comment]
This is not expected.  I try to reproduce the issue using the following code.  But the distribution is always reproducible. 

```
  program main
  use amrex_base_module
  implicit none
  integer ierr, i
  type(amrex_box) :: bx
  type(amrex_boxarray) :: ba
  type(amrex_distromap) :: dm
  type(amrex_multifab) :: mf
  call mpi_init(ierr)
  do i = 1, 10
     call amrex_init()
     if (amrex_parallel_ioprocessor()) then
        print *, "Hello world!"
     end if
     bx = amrex_box([0,0,0],[31,47,47])
     call amrex_boxarray_build(ba, bx)
     call ba%maxSize(16)
     call amrex_distromap_build(dm, ba)
     call amrex_multifab_build(mf, ba, dm, 1, 1)
     call amrex_print(dm)
     call amrex_multifab_destroy(mf)
     call amrex_boxarray_destroy(ba)
     call amrex_distromap_destroy(dm)
     call amrex_finalize()
  end do
  call mpi_finalize()
end program main
```

**Comment by nncarlson** [comment]
Okay, if it's not the expected behavior let me see if I can strip things down to an example that exhibits the irreproducibility. I took a quick stab with something simple like yours earlier without any luck. May be some sort of race condition.

**Comment by asalmgren** [comment]
@nncarlson -- just wanted to check in -- are you still seeing non-reproducible results or did this get resolved?  Thanks!

**Comment by nncarlson** [comment]
No, we're still seeing (reliably) non-reproducible results. I think we may be close to a smallish reproducer, which I we'll get back to in a few weeks after we get past some ECP reviews.

**Comment by asalmgren** [comment]
This should be interesting to track down.  Let us know when you have something reproducibly non-reproducible we can play with :-)
